gptme/message.py

```diff
@@ -352,9 +335,9 @@ def toml_to_msgs(toml: str) -> list[Message]:
     ]
 
 
-def msgs2dicts(msgs: list[Message], provider: Provider) -> list[dict]:
+def msgs2dicts(msgs: list[Message]) -> list[dict]:
     """Convert a list of Message objects to a list of dicts ready to pass to an LLM."""
-    return [msg.to_dict(keys=["role", "content"], provider=provider) for msg in msgs]
+    return [msg.to_dict(keys=["role", "content"]) for msg in msgs]
```

Comment:
with the other suggested changes, removing `provider=` here becomes the right call, but we need to let through `files`.

```py
def msgs2dicts(msgs: list[Message]) -> list[dict]:
    """Convert a list of Message objects to a list of dicts to be processed by providers"""
    return [msg.to_dict(keys=["role", "content", "files"]) for msg in msgs]
```